packages/server/package.json

```diff
@@ -10,22 +10,23 @@
   },
   "scripts": {
     "prebuild": "rimraf dist/",
-    "build": "tsc -p tsconfig.build.json && mv dist/src/* dist/ && rimraf dist/src/",
+    "build": "node ./scripts/build.js",
+    "postbuild": "copyfiles -f ../client/dist/budibase-client.js ../client/manifest.json client",
+    "check:types": "tsc -p tsconfig.build.json --noEmit",
     "build:dev": "yarn prebuild && tsc --build --watch --preserveWatchOutput",
     "debug": "yarn build && node --expose-gc --inspect=9222 dist/index.js",
-    "postbuild": "copyfiles -u 1 src/**/*.svelte dist/ && copyfiles -u 1 src/**/*.hbs dist/ && copyfiles -u 1 src/**/*.json dist/",
     "test": "bash scripts/test.sh",
     "test:memory": "jest --maxWorkers=2 --logHeapUsage --forceExit",
     "test:watch": "jest --watch",
-    "predocker": "copyfiles -f ../client/dist/budibase-client.js ../client/manifest.json client",
-    "build:docker": "yarn run predocker && docker build . -t app-service --label version=$BUDIBASE_RELEASE_VERSION",
+    "predocker": "copyfiles -f ../client/dist/budibase-client.js ../client/manifest.json client && yarn build  --configuration=production",
```

Comment:
Just to note for CI, we'll have already ran `yarn build --configuration=production` as a part of the pipeline definition. I'd be in favour of what you've done here, by keeping it inside the scripts definition instead. The only complication is the npm publish that done between the pipeline build and this docker build. Not sure what the best approach would be to prevent the need for this. 

Would the nx cache behaviour result in an almost instant build on this second step, or will the npm release mean it'll re-do a full build?

Reply:
As you mentioned the two build are required for different reasons, but no change should happen between both build. Nx should make it instant, so it should not be a problem.
The only change on the code should happen before the first build (Update versions)